### ISSUE TITLE
4 Cameras Not Found

### ISSUE THREAD

**Issue by jlpoolen** [lumeohq/onvif-rs#48]
I have four Reolink cameras on the network.  Verified that they respond to ONVIF broadcasts using another project's code.  (My cameras are Reolink which state they are ONVIf compliant; however, Reolink does not appear on the ONVIF web site, so I am not certain they are ONVIF _certified_.)

     jlpoole@taurus /usr/local/src/onvif-rs $ cargo run --example discovery
        Finished dev [unoptimized + debuginfo] target(s) in 0.07s
        Running `target/debug/examples/discovery`
     jlpoole@taurus /usr/local/src/onvif-rs $

**Comment by scottlamb** [comment]
`onvif-rs/examples/discovery.rs` waits for 1 second for reply. I found that wasn't long enough for some of my cameras. It might be interesting to try changing the `from_secs(1)` to `from_secs(5)`. And/or watching traffic with Wireshark.

**Comment by jlpoolen** [comment]
I tried the suggestions of @scottlamb  using values of 5 and then 20.  No difference.  I then ran as root; the whole project recompiled (4 m, 20 sec) and the output was the same: nothing.  I ran wireshark during the various attempts. The last attempt was with root for 5 seconds and wireshark ("tshark --color -i [interface]") posted 56 events.  I'd post the wireshark results, but I do not want to reveal anything outside of this issue/program.  I think I'm looking for UDP broadcasts and then any responses thereto.  The four cameras that are online did not appear to respond, there were no entries from their respective IPs, but I have to look closer at the wireshark capture log.  I'll update when I have a better command over analyzing the wireshark results.  Thanks Scott!

**Comment by jlpoolen** [comment]
The program that I used to confirm that my Reolink camera respond to an ONVIF broadcast is the Node.js project "Camera-probe" at https://github.com/patrickmichalina/camera-probe.  Camera-probe duly sent out UDP broadcasts on port 3702 and my cameras responded.  I did not find similar broadcasts in this project.  I'll study this code some to determine what is being sent out from where.

**Comment by jlpoolen** [comment]
There is a difference in the payload of the broadcast XML:

     Note: prettified = XML formatted with additional white space for readability

     1) Onvif-rs (Doesn't Work):

     No.	Time	Source	Destination	Protocol	Length	Info
     38	2.140237751	192.168.1.28	239.255.255.250	UDP	520	51779 → 3702 Len=478

     Data [prettified]:
     ^ÿúÜ¢f>oEú_«@fÀ¨ïÿÿúÊCvæ³¶<?xml version="1.0" encoding="utf-8"?>
     <s:Envelope xmlns:d="http://schemas.xmlsoap.org/ws/2005/04/discovery"  
                 xmlns:s="http://www.w3.org/2003/05/soap-envelope"
                 xmlns:w="http://schemas.xmlsoap.org/ws/2004/08/addressing">
	     <s:Header>
		     <w:MessageID>uuid:c29ac4a2-0d64-46ac-86ef-fb4516242589</w:MessageID>
	     <w:To>urn:schemas-xmlsoap-org:ws:2005:04:discovery</w:To>
	     <w:Action>http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe</w:Action>
               </s:Header>
           <s:Body>
		     <d:Probe />
     	</s:Body>
     </s:Envelope>

     2) camera-probe (Works):

     No.	Time	Source	Destination	Protocol	Length	Info
     3	2.834223124	192.168.1.28	239.255.255.250	UDP	828	49740 → 3702 Len=786

     Data [prettified]:
     ^ÿúÜ¢f>oE.¤x@ À¨ïÿÿúÂLv´ê<?xml version="1.0" encoding="UTF-8"?>
     <Envelope xmlns="http://www.w3.org/2003/05/soap-envelope">
       <Header xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing">
         <a:Action mustUnderstand="1">http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe</a:Action>     
         <a:MessageID>uuid:33b853b6-86d5-440f-9628-ae401a178d90</a:MessageID>
    <a:ReplyTo>
      <a:Address>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:Address>
    </a:ReplyTo>
    <a:To mustUnderstand="1">urn:schemas-xmlsoap-org:ws:2005:04:discovery</a:To>
       </Header>
       <Body>
               <Probe xmlns="http://schemas.xmlsoap.org/ws/2005/04/discovery">
                    <Types xmlns:dp0="http://www.onvif.org/ver10/network/wsdl">
               dp0:NetworkVideoTransmitter
                  </Types>
               </Probe>
          </Body>
     </Envelope>

**Comment by jlpoolen** [comment]
Here is a further analysis of the successful camera-probe:

     Summary:
    camera-probe sent out three UDP discovery request, each a unique Probe type.  The Reolink camera (#181) responded to all three discovery requests:

    From laptop running camera-probe:

    Broadcast#   Packet#  Probe-Type                MessageID
    1)           3        NetworkVideoTransmitter   uuid:33b853b6-86d5-440f-9628-ae401a178d90
    2)           4        Device                    uuid:0957592b-ab5e-45f6-9d70-0a729281e51b
    3)           5        NetworkVideoDisplay       uuid:4486e8c4-b2f7-42d9-b85a-c9531dcc6745


    Data:

    <?xml version="1.0" encoding="UTF-8"?>
    <Envelope xmlns="http://www.w3.org/2003/05/soap-envelope">
      <Header xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing">
        <a:Action mustUnderstand="1">http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe</a:Action>
        <a:MessageID>uuid:33b853b6-86d5-440f-9628-ae401a178d90</a:MessageID>
        <a:ReplyTo>
          <a:Address>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:Address>
        </a:ReplyTo>
        <a:To mustUnderstand="1">urn:schemas-xmlsoap-org:ws:2005:04:discovery</a:To>
      </Header>
      <Body>
        <Probe xmlns="http://schemas.xmlsoap.org/ws/2005/04/discovery">
          <Types xmlns:dp0="http://www.onvif.org/ver10/network/wsdl">
            dp0:NetworkVideoTransmitter
          </Types>
        </Probe>
      </Body>
    </Envelope>
    <?xml version="1.0" encoding="UTF-8"?>
    <Envelope xmlns="http://www.w3.org/2003/05/soap-envelope">
      <Header xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing">
        <a:Action mustUnderstand="1">http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe</a:Action>
        <a:MessageID>uuid:0957592b-ab5e-45f6-9d70-0a729281e51b</a:MessageID>
        <a:ReplyTo>
          <a:Address>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:Address>
        </a:ReplyTo>
        <a:To mustUnderstand="1">urn:schemas-xmlsoap-org:ws:2005:04:discovery</a:To>
      </Header>
      <Body>
        <Probe xmlns="http://schemas.xmlsoap.org/ws/2005/04/discovery">
          <Types xmlns:dp0="http://www.onvif.org/ver10/network/wsdl">
            dp0:Device
          </Types>
        </Probe>
      </Body>
    </Envelope>
    <?xml version="1.0" encoding="UTF-8"?>
    <Envelope xmlns="http://www.w3.org/2003/05/soap-envelope">
      <Header xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing">
        <a:Action mustUnderstand="1">http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe</a:Action>
        <a:MessageID>uuid:4486e8c4-b2f7-42d9-b85a-c9531dcc6745</a:MessageID>
        <a:ReplyTo>
          <a:Address>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:Address>
        </a:ReplyTo>
        <a:To mustUnderstand="1">urn:schemas-xmlsoap-org:ws:2005:04:discovery</a:To>
      </Header>
      <Body>
        <Probe xmlns="http://schemas.xmlsoap.org/ws/2005/04/discovery">
          <Types xmlns:dp0="http://www.onvif.org/ver10/network/wsdl">
            dp0:NetworkVideoDisplay
          </Types>
        </Probe>
      </Body>
    </Envelope>

    Responses from Camera 181 to all 3 discovery probes:

    1) Response of Camera 181 in packets 6,7,8 (assembled in 9), responds to broadcast #1:

    <?xml version="1.0" encoding="UTF-8"?>
    <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope" xmlns:SOAP-ENC="http://www.w3.org/2003/05/soap-encoding" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsdd="http://schemas.xmlsoap.org/ws/2005/04/discovery" xmlns:wsa5="http://www.w3.org/2005/08/addressing" xmlns:xmime="http://tempuri.org/xmime.xsd" xmlns:xmime5="http://www.w3.org/2005/05/xmlmime" xmlns:xop="http://www.w3.org/2004/08/xop/include" xmlns:wsrfbf="http://docs.oasis-open.org/wsrf/bf-2" xmlns:tt="http://www.onvif.org/ver10/schema" xmlns:wstop="http://docs.oasis-open.org/wsn/t-1" xmlns:wsrfr="http://docs.oasis-open.org/wsrf/r-2" xmlns:ns1="http://www.onvif.org/ver10/actionengine/wsdl" xmlns:tev="http://www.onvif.org/ver10/events/wsdl" xmlns:ns10="http://www.onvif.org/ver10/events/wsdl/PullPointBinding" xmlns:ns11="http://www.onvif.org/ver10/events/wsdl/CreatePullPointBinding" xmlns:ns12="http://www.onvif.org/ver10/events/wsdl/PausableSubscriptionManagerBinding" xmlns:ns13="http://www.onvif.org/ver10/network/wsdl/RemoteDiscoveryBinding" xmlns:ns14="http://www.onvif.org/ver10/network/wsdl/DiscoveryLookupBinding" xmlns:tdn="http://www.onvif.org/ver10/network/wsdl" xmlns:ns3="http://www.onvif.org/ver20/analytics/wsdl/RuleEngineBinding" xmlns:ns4="http://www.onvif.org/ver20/analytics/wsdl/AnalyticsEngineBinding" xmlns:tan="http://www.onvif.org/ver20/analytics/wsdl" xmlns:ns5="http://www.onvif.org/ver10/events/wsdl/PullPointSubscriptionBinding" xmlns:ns6="http://www.onvif.org/ver10/events/wsdl/EventBinding" xmlns:ns7="http://www.onvif.org/ver10/events/wsdl/SubscriptionManagerBinding" xmlns:ns8="http://www.onvif.org/ver10/events/wsdl/NotificationProducerBinding" xmlns:wsnt="http://docs.oasis-open.org/wsn/b-2" xmlns:ns9="http://www.onvif.org/ver10/events/wsdl/NotificationConsumerBinding" xmlns:tad="http://www.onvif.org/ver10/analyticsdevice/wsdl" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:timg="http://www.onvif.org/ver20/imaging/wsdl" xmlns:tls="http://www.onvif.org/ver10/display/wsdl" xmlns:tmd="http://www.onvif.org/ver10/deviceIO/wsdl" xmlns:tptz="http://www.onvif.org/ver20/ptz/wsdl" xmlns:trc="http://www.onvif.org/ver10/recording/wsdl" xmlns:trp="http://www.onvif.org/ver10/replay/wsdl" xmlns:trt="http://www.onvif.org/ver10/media/wsdl" xmlns:trv="http://www.onvif.org/ver10/receiver/wsdl" xmlns:tse="http://www.onvif.org/ver10/search/wsdl" xmlns:ter="http://www.onvif.org/ver10/error" xmlns:tns1="http://www.onvif.org/ver10/topics" xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd">
        <SOAP-ENV:Header>
            <wsa:MessageID>uuid:2419d68a-2dd2-21b2-a205-EC:71:DB:7D:15:92</wsa:MessageID>
            <wsa:RelatesTo>uuid:33b853b6-86d5-440f-9628-ae401a178d90</wsa:RelatesTo>
            <wsa:To SOAP-ENV:mustUnderstand="1">http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</wsa:To>
            <wsa:Action>http://schemas.xmlsoap.org/ws/2005/04/discovery/ProbeMatches</wsa:Action>
        </SOAP-ENV:Header>
        <SOAP-ENV:Body>
            <wsdd:ProbeMatches>
                <wsdd:ProbeMatch>
                    <wsa:EndpointReference>
                        <wsa:Address>urn:uuid:2419d68a-2dd2-21b2-a205-EC:71:DB:7D:15:92</wsa:Address>
                        <wsa:ReferenceProperties/>
                        <wsa:ReferenceParameters/>
                        <wsa:PortType>ttl</wsa:PortType>
                    </wsa:EndpointReference>
                    <wsdd:Types>tdn:NetworkVideoTransmitter</wsdd:Types>
                    <wsdd:Scopes> onvif://www.onvif.org/type/video_encoder onvif://www.onvif.org/location/country/china onvif://www.onvif.org/type/network_video_transmitter onvif://www.onvif.org/hardware/IPC-122 onvif://www.onvif.org/Profile/Streaming onvif://www.onvif.org/name/IPC-BO</wsdd:Scopes>
                    <wsdd:XAddrs>http://192.168.1.181:8000/onvif/device_service</wsdd:XAddrs>
                    <wsdd:MetadataVersion>1</wsdd:MetadataVersion>
                </wsdd:ProbeMatch>
            </wsdd:ProbeMatches>
        </SOAP-ENV:Body>
    </SOAP-ENV:Envelope>

    2) Response of packets 17,18, & 19 (assembled as packet #20) responds to broadcast #2:

    <?xml version="1.0" encoding="UTF-8"?>
    <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope" xmlns:SOAP-ENC="http://www.w3.org/2003/05/soap-encoding" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsdd="http://schemas.xmlsoap.org/ws/2005/04/discovery" xmlns:wsa5="http://www.w3.org/2005/08/addressing" xmlns:xmime="http://tempuri.org/xmime.xsd" xmlns:xmime5="http://www.w3.org/2005/05/xmlmime" xmlns:xop="http://www.w3.org/2004/08/xop/include" xmlns:wsrfbf="http://docs.oasis-open.org/wsrf/bf-2" xmlns:tt="http://www.onvif.org/ver10/schema" xmlns:wstop="http://docs.oasis-open.org/wsn/t-1" xmlns:wsrfr="http://docs.oasis-open.org/wsrf/r-2" xmlns:ns1="http://www.onvif.org/ver10/actionengine/wsdl" xmlns:tev="http://www.onvif.org/ver10/events/wsdl" xmlns:ns10="http://www.onvif.org/ver10/events/wsdl/PullPointBinding" xmlns:ns11="http://www.onvif.org/ver10/events/wsdl/CreatePullPointBinding" xmlns:ns12="http://www.onvif.org/ver10/events/wsdl/PausableSubscriptionManagerBinding" xmlns:ns13="http://www.onvif.org/ver10/network/wsdl/RemoteDiscoveryBinding" xmlns:ns14="http://www.onvif.org/ver10/network/wsdl/DiscoveryLookupBinding" xmlns:tdn="http://www.onvif.org/ver10/network/wsdl" xmlns:ns3="http://www.onvif.org/ver20/analytics/wsdl/RuleEngineBinding" xmlns:ns4="http://www.onvif.org/ver20/analytics/wsdl/AnalyticsEngineBinding" xmlns:tan="http://www.onvif.org/ver20/analytics/wsdl" xmlns:ns5="http://www.onvif.org/ver10/events/wsdl/PullPointSubscriptionBinding" xmlns:ns6="http://www.onvif.org/ver10/events/wsdl/EventBinding" xmlns:ns7="http://www.onvif.org/ver10/events/wsdl/SubscriptionManagerBinding" xmlns:ns8="http://www.onvif.org/ver10/events/wsdl/NotificationProducerBinding" xmlns:wsnt="http://docs.oasis-open.org/wsn/b-2" xmlns:ns9="http://www.onvif.org/ver10/events/wsdl/NotificationConsumerBinding" xmlns:tad="http://www.onvif.org/ver10/analyticsdevice/wsdl" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:timg="http://www.onvif.org/ver20/imaging/wsdl" xmlns:tls="http://www.onvif.org/ver10/display/wsdl" xmlns:tmd="http://www.onvif.org/ver10/deviceIO/wsdl" xmlns:tptz="http://www.onvif.org/ver20/ptz/wsdl" xmlns:trc="http://www.onvif.org/ver10/recording/wsdl" xmlns:trp="http://www.onvif.org/ver10/replay/wsdl" xmlns:trt="http://www.onvif.org/ver10/media/wsdl" xmlns:trv="http://www.onvif.org/ver10/receiver/wsdl" xmlns:tse="http://www.onvif.org/ver10/search/wsdl" xmlns:ter="http://www.onvif.org/ver10/error" xmlns:tns1="http://www.onvif.org/ver10/topics" xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd">
        <SOAP-ENV:Header>
            <wsa:MessageID>uuid:2419d68a-2dd2-21b2-a205-EC:71:DB:7D:15:92</wsa:MessageID>
            <wsa:RelatesTo>uuid:0957592b-ab5e-45f6-9d70-0a729281e51b</wsa:RelatesTo>
            <wsa:To SOAP-ENV:mustUnderstand="1">http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</wsa:To>
            <wsa:Action>http://schemas.xmlsoap.org/ws/2005/04/discovery/ProbeMatches</wsa:Action>
        </SOAP-ENV:Header>
        <SOAP-ENV:Body>
            <wsdd:ProbeMatches>
                <wsdd:ProbeMatch>
                    <wsa:EndpointReference>
                        <wsa:Address>urn:uuid:2419d68a-2dd2-21b2-a205-EC:71:DB:7D:15:92</wsa:Address>
                        <wsa:ReferenceProperties/>
                        <wsa:ReferenceParameters/>
                        <wsa:PortType>ttl</wsa:PortType>
                    </wsa:EndpointReference>
                    <wsdd:Types>tdn:NetworkVideoTransmitter</wsdd:Types>
                    <wsdd:Scopes> onvif://www.onvif.org/type/video_encoder onvif://www.onvif.org/location/country/china onvif://www.onvif.org/type/network_video_transmitter onvif://www.onvif.org/hardware/IPC-122 onvif://www.onvif.org/Profile/Streaming onvif://www.onvif.org/name/IPC-BO</wsdd:Scopes>
                    <wsdd:XAddrs>http://192.168.1.181:8000/onvif/device_service</wsdd:XAddrs>
                    <wsdd:MetadataVersion>1</wsdd:MetadataVersion>
                </wsdd:ProbeMatch>
            </wsdd:ProbeMatches>
        </SOAP-ENV:Body>
    </SOAP-ENV:Envelope>

    3) Response of packets 27,28, & 29 (reassembled in 30) responds to broadcast #3:

    <?xml version="1.0" encoding="UTF-8"?>
    <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope" xmlns:SOAP-ENC="http://www.w3.org/2003/05/soap-encoding" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsdd="http://schemas.xmlsoap.org/ws/2005/04/discovery" xmlns:wsa5="http://www.w3.org/2005/08/addressing" xmlns:xmime="http://tempuri.org/xmime.xsd" xmlns:xmime5="http://www.w3.org/2005/05/xmlmime" xmlns:xop="http://www.w3.org/2004/08/xop/include" xmlns:wsrfbf="http://docs.oasis-open.org/wsrf/bf-2" xmlns:tt="http://www.onvif.org/ver10/schema" xmlns:wstop="http://docs.oasis-open.org/wsn/t-1" xmlns:wsrfr="http://docs.oasis-open.org/wsrf/r-2" xmlns:ns1="http://www.onvif.org/ver10/actionengine/wsdl" xmlns:tev="http://www.onvif.org/ver10/events/wsdl" xmlns:ns10="http://www.onvif.org/ver10/events/wsdl/PullPointBinding" xmlns:ns11="http://www.onvif.org/ver10/events/wsdl/CreatePullPointBinding" xmlns:ns12="http://www.onvif.org/ver10/events/wsdl/PausableSubscriptionManagerBinding" xmlns:ns13="http://www.onvif.org/ver10/network/wsdl/RemoteDiscoveryBinding" xmlns:ns14="http://www.onvif.org/ver10/network/wsdl/DiscoveryLookupBinding" xmlns:tdn="http://www.onvif.org/ver10/network/wsdl" xmlns:ns3="http://www.onvif.org/ver20/analytics/wsdl/RuleEngineBinding" xmlns:ns4="http://www.onvif.org/ver20/analytics/wsdl/AnalyticsEngineBinding" xmlns:tan="http://www.onvif.org/ver20/analytics/wsdl" xmlns:ns5="http://www.onvif.org/ver10/events/wsdl/PullPointSubscriptionBinding" xmlns:ns6="http://www.onvif.org/ver10/events/wsdl/EventBinding" xmlns:ns7="http://www.onvif.org/ver10/events/wsdl/SubscriptionManagerBinding" xmlns:ns8="http://www.onvif.org/ver10/events/wsdl/NotificationProducerBinding" xmlns:wsnt="http://docs.oasis-open.org/wsn/b-2" xmlns:ns9="http://www.onvif.org/ver10/events/wsdl/NotificationConsumerBinding" xmlns:tad="http://www.onvif.org/ver10/analyticsdevice/wsdl" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:timg="http://www.onvif.org/ver20/imaging/wsdl" xmlns:tls="http://www.onvif.org/ver10/display/wsdl" xmlns:tmd="http://www.onvif.org/ver10/deviceIO/wsdl" xmlns:tptz="http://www.onvif.org/ver20/ptz/wsdl" xmlns:trc="http://www.onvif.org/ver10/recording/wsdl" xmlns:trp="http://www.onvif.org/ver10/replay/wsdl" xmlns:trt="http://www.onvif.org/ver10/media/wsdl" xmlns:trv="http://www.onvif.org/ver10/receiver/wsdl" xmlns:tse="http://www.onvif.org/ver10/search/wsdl" xmlns:ter="http://www.onvif.org/ver10/error" xmlns:tns1="http://www.onvif.org/ver10/topics" xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd">
        <SOAP-ENV:Header>
            <wsa:MessageID>uuid:2419d68a-2dd2-21b2-a205-EC:71:DB:7D:15:92</wsa:MessageID>
            <wsa:RelatesTo>uuid:4486e8c4-b2f7-42d9-b85a-c9531dcc6745</wsa:RelatesTo>
            <wsa:To SOAP-ENV:mustUnderstand="1">http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</wsa:To>
            <wsa:Action>http://schemas.xmlsoap.org/ws/2005/04/discovery/ProbeMatches</wsa:Action>
        </SOAP-ENV:Header>
        <SOAP-ENV:Body>
            <wsdd:ProbeMatches>
                <wsdd:ProbeMatch>
                    <wsa:EndpointReference>
                        <wsa:Address>urn:uuid:2419d68a-2dd2-21b2-a205-EC:71:DB:7D:15:92</wsa:Address>
                        <wsa:ReferenceProperties/>
                        <wsa:ReferenceParameters/>
                        <wsa:PortType>ttl</wsa:PortType>
                    </wsa:EndpointReference>
                    <wsdd:Types>tdn:NetworkVideoTransmitter</wsdd:Types>
                    <wsdd:Scopes> onvif://www.onvif.org/type/video_encoder onvif://www.onvif.org/location/country/china onvif://www.onvif.org/type/network_video_transmitter onvif://www.onvif.org/hardware/IPC-122 onvif://www.onvif.org/Profile/Streaming onvif://www.onvif.org/name/IPC-BO</wsdd:Scopes>
                    <wsdd:XAddrs>http://192.168.1.181:8000/onvif/device_service</wsdd:XAddrs>
                    <wsdd:MetadataVersion>1</wsdd:MetadataVersion>
                </wsdd:ProbeMatch>
            </wsdd:ProbeMatches>
        </SOAP-ENV:Body>
    </SOAP-ENV:Envelope>

**Comment by chrisabruce** [comment]
@jlpoolen this issue may be resolved now with latest merge.  If you can retest with master, that would be helpful.

**Comment by jlpoolen** [comment]
Initial tests look good, I'm able to fetch the date/time and information.  

It's been so long since I had this in my mind, I've had to re-familiarize myself and got hung up on getting Wireshark (both in Linux and Windows) to give me information I want -- I cheated and took a short-cut and used your ready-to-use commands, e.g.

     cargo run --example camera -- get-system-date-and-time  --uri=http://192.168.1.XX:8000

and got meaningful XML back.

Just for my sake, and others looking at this in the future, I'll try to document in a bit more detail some Wireshark debugging tips  as using Wireshark in conjunction with this software gives one a powerful command of knowing what is happening.

Thank you!
